### ISSUE TITLE
spray object with zero entries

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/spray#2]
Under some circumstances we can have a spray object with zero entries:

```
> spray(matrix(c(1,1,2,2),2,2),c(1,-1),addrepeats=TRUE)
         val
 1 2  =    0
> 
```

**Comment by RobinHankin** [comment]
Observe that the following ```C++``` idiom

```
mycont v;
spray::iterator it;

for(it=S.begin(); it != S.end(); /* nop (sic) */){
  if(it->second == 0){ S.erase(it->first); }
 }
```

is buggy because we are iremoving entries from a map while iterating over it.

**Comment by RobinHankin** [comment]
fixed in 5a53b1e
